bftengine/src/bftengine/messages/PrePrepareMsg.cpp

```diff
@@ -194,6 +196,29 @@ bool PrePrepareMsg::checkRequests() const {
   Assert(false);
   return false;
 }
+std::string PrePrepareMsg::getMessageClientCorrelationId(int index) {
+  auto it = RequestsIterator(this);
+  int req_num = 0;
+  while (!it.end() && req_num < index) {
+    it.gotoNext();
+    req_num++;
+  }
+  char* requestBody = nullptr;
+  it.getCurrent(requestBody);
```

Comment:
If `it.end() == true`, I think we will have a segfault. Since it is a public method, we can either document what the index should be as a precondition (and assert on it) or we should handle the end case in some other way - maybe an optional return or an exception.

Reply:
fixed